pwa/src/services/stringValidations.ts

```diff
@@ -20,8 +20,13 @@ export const validateThrowsArray = (rawString: string): boolean | string => {
   }
 };
 
-export const validatePassword = (value: string, validationValue: string): string | boolean => {
-  if (value === validationValue || (!value && !validationValue)) return true;
+export const validatePassword = (value: string, validationValue: string, user?: any): string | boolean => {
+  if (user) {
+    if (value === validationValue || (!value && !validationValue)) return true;
+  } else {
+    if (!value && !validationValue) return "Password is required";
+    if (value && validationValue && value === validationValue) return true;
+  }
```

Comment:
This function is quite complex and doesn't exactly do what we want it to do (e.g. see screenshot, but there are other issues as well).

I'm thinking one of the main issues here is that there are _two_ inputs that are dependent on each other, and react-hook-form is not updating both of them when necessary.

Maybe we could pair-program a solution for this? (I don't think we should merge this as-is).

![image](https://user-images.githubusercontent.com/66552932/222367254-22abc749-abb6-4ac8-adc8-42b7af49c021.png)